lib/jxl/modular/encoding/encoding.cc

```diff
@@ -403,6 +404,13 @@ Status ModularDecode(BitReader *br, Image &image, GroupHeader &header,
   }
 
   size_t nb_channels = image.channel.size();
+  size_t next_channel = 0;
+  auto scope_guard = MakeScopeGuard([&]() {
```

Comment:
In the case where `num_chans == 0` in the code below (which I think might happen e.g. when decoding modular DC groups in a large image that doesn't do squeeze, so the DC groups are empty), starting the scope guard here will cause all channels to get zero-initialized while that is not really necessary in such a case.

Maybe start the scope guard after the line with `if (num_chans == 0) return true;`?

Reply:
Sure! Thanks.